astropy/modeling/fitting.py

```diff
@@ -1131,9 +1135,12 @@ class _NonLinearLSQFitter(Fitter):
     The constraint types supported by this fitter type.
     """
 
-    def __init__(self, calc_uncertainties=False, use_min_max_bounds=True):
+    def __init__(
+        self, calc_uncertainties=False, use_min_max_bounds=True, check_non_finite=True
```

Comment:
![Screenshot 2025-03-06 at 10 45 49](https://github.com/user-attachments/assets/1b70e77d-ad97-4432-a1f9-04938bbadf89)
[boolean traps](https://adamj.eu/tech/2021/07/10/python-type-hints-how-to-avoid-the-boolean-trap/)
```suggestion
        self, calc_uncertainties=False, use_min_max_bounds=True, *, check_non_finite=True
```

Reply:
I don't see the point of your article and image here.
Let's see if modeling maintainers want to start using keyword only arguments but I think this is something we should do in major release and just enforce all keyword args to be keyword only. Otherwise it just adds noise and make weird signatures.

Reply:
To suddenly enforce keyword only would break a lot of downstream code.  I'd rather not. Since this new keyword is at the end of the existing signature, it is currently backward compatible.

Reply:
> To suddenly enforce keyword only would break a lot of downstream code. 

Not for a *new* argument; nobody is currently passing it positionally since it's not a thing yet.

> I don't see the point of your article and image here.

Sorry I should have been more explicit. My point is that keyword-only argument are much easier to maintain (deprecating an argument that is allowed positionally is basically not feasible without subtle breaks). Making *new* arguments keyword-only from the get go has no cost in terms of backward compatibility, but changing them to be keyword-only later on does: as you pointed out, it cannot reasonably be done outside of a major release, and even then, it may be argued that it's not worth the churn.
Allowing boolean argument positionally is also problematic in most cases because callsites can quickly become unclear.
The reason I posted a meme instead of an explanation is that this must be the single most common design mistake I see in my reviews, and I used Adam Johnson's article as a way to avoid repeating this explanation for the nth time. This was unprofessional and a mistake, I apologize for not communicating more clearly.

Reply:
I would prefer the new keyword argument be made keyword only as suggested.